cli/cliui/agent.go

```diff
@@ -389,15 +391,32 @@ func ConnDiagnostics(w io.Writer, d ConnDiags) {
 		return
 	}
 
-	if d.ConnInfo != nil && d.ConnInfo.DERPMap != nil && !d.ConnInfo.DERPMap.HasSTUN() {
-		_, _ = fmt.Fprint(w, "✘ The DERP map is not configured to use STUN, which will prevent direct connections from starting outside of local networks\n")
+	if d.ConnInfo != nil && d.ConnInfo.DERPMap != nil {
+		if !d.ConnInfo.DERPMap.HasSTUN() {
+			_, _ = fmt.Fprint(w, "✘ The DERP map is not configured to use STUN, which will prevent direct connections from starting outside of local networks\n")
+		} else if d.LocalNetInfo != nil && !d.LocalNetInfo.UDP {
+			_, _ = fmt.Fprint(w, "❗ Client could not connect to STUN over UDP, which may be preventing a direct connection\n")
```

Comment:
The [notion doc](https://www.notion.so/coderhq/P2P-Diagnostics-fffd579be59280569e15d26ae2eaa324) on this proposed organizing agent problems and client problems into their own sections, and having a preamble that explains these are problems with direct connections.  I think that's preferable to putting `Client | Agent` on each and having them unordered, and having to put "which may be preventing a direct connection" as a suffix a bunch of times.